File: src/jp/lionas/alexa/fundra/SlotUtil.java
package jp.lionas.alexa.fundra;

import com.amazon.ask.dispatcher.request.handler.HandlerInput;
import com.amazon.ask.model.Intent;
import com.amazon.ask.model.IntentRequest;
import com.amazon.ask.model.Request;
import com.amazon.ask.model.Slot;

import java.util.Map;

class SlotUtil {

    static Map<String, Slot> getSlots(HandlerInput handlerInput) {

        Request request = handlerInput.getRequestEnvelope().getRequest();
        IntentRequest intentRequest = (IntentRequest) request;
        Intent intent = intentRequest.getIntent();
        return intent.getSlots();

    }

}

File: src/jp/lionas/alexa/fundra/util/SlotUtil.kt
package jp.lionas.alexa.fundra.util

import com.amazon.ask.dispatcher.request.handler.HandlerInput
import com.amazon.ask.model.IntentRequest
import com.amazon.ask.model.Slot

internal class SlotUtil {

    companion object {

        @JvmStatic
        fun getSlots(handlerInput: HandlerInput): MutableMap<String, Slot>? {

            val request = handlerInput.requestEnvelope.request
            val intentRequest = request as IntentRequest?
            val intent = intentRequest?.intent

            return intent?.slots

        }

    }

}
